### PR TITLE
Bump firebase version in /functions and in main project

### PR DESCRIPTION
**Pull request by yurii-lubynets**
`8.8.0`

Before merging these changes it'd be tested via [preview link](https://co-reality-staging--preview-pr-1883-zi6xqu3a.web.app/in/devaliashacksville)

related to https://github.com/sparkletown/internal-sparkle-issues/issues/938

### Release Notes

- **Old Version:** 7.14.2 (April 23, 2020): [release notes](https://firebase.google.com/support/release-notes/js#version_7142_-_april_23_2020)
- _a whole bunch of bug fixes, features, etc_ (only calling out some interesting/notable ones)
  - `7.14.6` (May 29, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7146_-_may_29_2020))
    - > Feature: Added support for calling `FirebaseFiresore.settings` with `{ ignoreUndefinedProperties: true }`. When this parameter is set, Cloud Firestore ignores undefined properties inside objects rather than rejecting the API call.
  - `7.16.0` (July 09, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7160_-_july_09_2020))
    - > Fixed: Clear timeout after a successful response or after the request is canceled
      - > Callable Function throw Deadline Exceeded Error even on successful response
  - `7.18.0` (August 13, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7180_-_august_13_2020))
    - > Fixed: The SDK no longer crashes with the error "The database connection is closing". Instead, the individual operations that cause this error may be rejected.
  - `7.19.0` (August 20, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7190_-_august_20_2020))
    - > Feature: Released `@firebase/rules-unit-testing` to replace the `@firebase/testing package`.
  - `7.21.0` (September 17, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7210_-_september_17_2020))
    - > Changed: The SDK now uses `FirestoreError` instead of Error in `onSnapshot*()` error callbacks.
  - `7.22.0` (October 1, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7220_-_october_1_2020))
    - > Feature: Users can now set a custom domain for callable functions.
  - `7.23.0` (October 8, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_7230_-_october_8_2020))
    - > Changed: Moved the `loggingEnabled` check to wait until performance initialization finishes, **which prevents the SDK from dropping custom traces** right after getting the performance object.
    - > Throws an exception when `startTime` or duration is not a positive value in the `trace.record()` API.
- `8.0.0` (October 26, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_800_-_october_26_2020), **BREAKING CHANGES**)
  - > Breaking change: browser fields in `package.json` files now point to ESM bundles instead of CJS bundles. Users who are using ESM imports must now use the default import instead of a namespace import.
    - > Before 8.0.0: `import * as firebase from 'firebase/app'`
    - > After 8.0.0: `import firebase from 'firebase/app'`
    - > Code that uses `require('firebase/app')` or `require('firebase')` will still work, but in order to get proper typings (for code completion, for example) users should change these require calls to `require('firebase/app').default` or `require('firebase').default`. This is because the SDK now uses typings for the ESM bundle, and the different bundles share one typings file.
  - > Breaking change: Removed deprecated `experimentalTabSynchronization` settings. To enable multi-tab synchronization, use `synchronizeTabs` instead.
  - > Breaking change: Removed the `timestampsInSnapshots` option from `FirestoreSettings`. Now, Cloud Firestore always returns `Timestamp` values for all timestamp values.
  - > Changed: This release removes runtime type validations that are covered by TypeScript. Developers can use our TypeScript types to validate API usage.
- _a whole bunch of bug fixes, features, etc_ (only calling out some interesting/notable ones)
  - `8.0.1` (November 5, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_801_-_november_5_2020))
    - > Feature: Added a `withFunctionTriggersDisabled` method which runs a user-provided setup function with emulated Cloud Functions triggers disabled. This can be used to import data into the Realtime Database or Cloud Firestore emulators without triggering locally emulated functions. This method only works with Firebase CLI version 8.13.0 or higher.
  - `8.0.2` (November 12, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_802_-_november_12_2020))
    - > The SDK now retries IndexedDB errors a fixed number of times to handle connection issues in mobile WebKit.
  - `8.1.2` (; [release notes](https://firebase.google.com/support/release-notes/js#version_812_-_december_3_2020))
    - > Fixed: Fixed a bug that prevented usage of `FieldPath` objects with multiple special characters.
  - `8.2.0` (December 11, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_820_-_december_11_2020))
    - > Feature: Released Firestore Bundles (pre-packaged Firestore data). For NPM users, this can be enabled via an additional import: `firebase/firestore/bundle`. For CDN usage, it is enabled by default.
    - > Changed: A write to a document that contains `FieldValue` transforms is no longer split up into two separate operations. This reduces the number of writes the backend performs and allows each `WriteBatch` to hold `500` writes regardless of how many `FieldValue` transformations are attached.
    - > Changed: Changed to dispatch no more than `1000` collected performance events for each network request. Events over the cap will be queued and sent with the next dispatch.
  - `8.2.1` (December 17, 2020; [release notes](https://firebase.google.com/support/release-notes/js#version_821_-_december_17_2020))
    - > Fixed: Fixed an issue that prevented the SDK from automatically retrieving custom claims for a `User`
  - `8.2.2` (January 7, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_822_-_january_7_2021))
    - > Fixed: Fixed an issue in the Transaction API that caused the SDK to return invalid DocumentReference objects through DocumentSnapshot.data() calls.
  - `8.2.4` (January 21, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_824_-_january_21_2021))
    - > Fixed: Fixed an error that caused `FirestoreDataConverter.fromFirestore()` to be called with an incorrect `QueryDocumentSnapshot` object.
  - `8.2.5` (January 28, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_825_-_january_28_2021))
    - > Fixed: Classes like `DocumentReference` and `Query` can now be serialized to `JSON`
  - `8.2.6` (February 04, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_826_-_february_04_2021))
    - > Fixed: Correctly handle `ignoreUndefinedProperties` in `set({ merge: true })`. Previously this would behave as if the undefined value were `FieldValue.delete()`, which wasn't intended.
  - `8.2.7` (February 11, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_827_-_february_11_2021))
    - > Fixed a bug where local cache inconsistencies were unnecessarily being resolved.
  - `8.2.8` (February 18, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_828_-_february_18_2021))
    - > Fixed a behavior causing `gtag.js` to be downloaded twice on Firebase Analytics initialization. 
  - `8.3.0` (March 10, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_830_-_march_10_2021))
    - > Feature: Added support to remove the `FirestoreDataConverter` on a `Firestore` reference by calling `withConverter(null)`
  - `8.4.0` (April 12, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_840_-_april_12_2021))
    - > Fixed a bug where decimal inputs to `Timestamp.fromMillis()` were calculated incorrectly due to floating point precision loss.
  - `8.4.2` (April 23, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_842_-_april_23_2021))
    - > Fixed an issue where errors from `grpc` are thrown directly to user code. Now they are wrapped in `FirestoreError`.
    - > Fixed the infinite recursion caused by the `FirebaseStorageError` message getter.
  - `8.5.0` (May 5, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_850_-_may_5_2021))
    - > `useEmulator` now accepts an optional mockUserToken
  - `8.6.0` (May 11, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_860_-_may_11_2021))
    - > Published the App Check package
    - > Fixed typings for useEmulator
    - > Added support for Cloud Storage emulator.
  - `8.6.1` (May 12, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_861_-_may_12_2021))
    - > Fixed a bug that caused crashes in Cloud Storage, Cloud Functions for Firebase Client SDK, and Realtime Database when using the all-in-one firebase import, as in `import firebase from 'firebase'`.
    - > Fixed an issue that could cause `once()` to fire more than once if the value was modified inside its callback.
  - `8.6.2` (May 20, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_862_-_may_20_2021))
    - > Use `pagehide` event for page termination by default, instead of unload. [See docs on pagehide vs unload](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#legacy-lifecycle-apis-to-avoid).
    - > Fixed a bug in `httpsCallable()` when it is used in the same project as Cloud Messaging. See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/4909.
    - > Prevent `useEmulators()` from erroring when only the `storage` field is populated in the options parameter.
  - `8.6.3` (May 27, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_863_-_may_27_2021))
    - > Fixed an "index not defined" error that sometimes occurred during a get() call on an orderByChild query.
    - > Added a warning message when settings() is used to override the original value of host without also setting {merge: true}.
  - `8.6.5` (June 4, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_865_-_june_4_2021))
    - > Fixed an error where `firebase.SDK_VERSION` did not always contain the correct version number.
    - > Fixed an issue where `FetchXmlHttpFactory` was not available.
  - `8.6.7` (June 10, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_865_-_june_10_2021))
    - > Fixed a regression that prevented Cloud Firestore from detecting the Authentication service during initialization, which prevented some writes from being sent.
  - `8.6.8` (June 17, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_868_-_june_17_2021))
    - > Fixed a bug where linkWithPopup, linkWithRedirect, reauthenticateWithPopup, and reauthenticateWithRedirect weren't correctly picking up the emulator configuration. See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/4879.
    - > Fixed a bug that caused Uncaught TypeError: e.auth.addAuthTokenListener is not a function. See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/5018.
  - `8.7.0` (July 1, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_870_-_july_1_2021))
    - > Added a getToken() method that returns an App Check token asynchronously.
    - > Added an onTokenChanged() method that adds listeners to App Check token changes.
    - > Fixed an error causing HTTP status 429 errors in debug mode. See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/5052.
    - > Fixed an issue where Cloud Firestore didn't detect Authentication, which led to permission denied errors. See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/4932.
  - `8.7.1` (July 8, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_871_-_july_8_2021))
    - > In Node, always send Authentication and App Check tokens when they are available. This fixes a bug affecting Admin SDK users who have App Check enforced.
  - `8.8.0` (July 22, 2021; [release notes](https://firebase.google.com/support/release-notes/js#version_880_-_july_22_2021))
    - > Added firebase_screen and firebase_screen_class to logEvent() overload for screen_view events. These fields are required to log screen_view events correctly to Analytics.
    - > Fixed sending of Authentication tokens in Node.js. This bug affected Admin SDK users who have App Check enforced.
    - > The SDK no longer accesses IndexedDB during a page unload event on Safari 14. This aims to reduce the occurrence of an IndexedDB bug in Safari (https://bugs.webkit.org/show_bug.cgi?id=226547). See GitHub issue https://github.com/firebase/firebase-js-sdk/issues/4076.
    - > Fixed an issue that prevented Date objects from being used via update() when connected to the Cloud Firestore emulator.
    - > Added Node.js support to Cloud Storage for Firebase. This release changes the main field in package.json to point to a Node-specific build. If you are building a bundle for browser usage, make sure that your bundler uses the browser field (this is usually the default).

### CODE REVIEW

**Comment by yurii-lubynets** [comment]
will be fixed in scope of https://github.com/sparkletown/sparkle/pull/1883

File: src/types/UpcomingEvent.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import type firebase from "firebase/app";
+
```
Comment (yurii-lubynets) [inline src/types/UpcomingEvent.ts:2]: Error fix `Namespace '"/Applications/projects/sparkle/sparkle/node_modules/firebase/index"' has no exported member 'firestore'. ts(2694)`
See details https://github.com/firebase/firebase-js-sdk/issues/4004

File: src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { useCallback, useMemo, useRef, useState } from "react";
 import { useFirebase } from "react-redux-firebase";
 import { useAsync } from "react-use";
-import { UserInfo } from "firebase/app";
+import firebase from "firebase/app";
```
Comment (yurii-lubynets) [inline src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx:4]: `Module '"firebase/app"' has no exported member 'UserInfo'. ts(2614)`

**Comment by 0xdevalias** [comment]
> > will be fixed in scope #1883
> 
> @yurii-lubynets it tends to be poor open-source etiquette to throw away previous contributions that solve the same problem; particularly without at least making use of the commits contributed, etc.
> 
> _Originally posted by @0xdevalias in https://github.com/sparkletown/sparkle/pull/1362#issuecomment-886171055_

**Pull request merged**